src/TS3Connection.ts

```diff
@@ -440,34 +457,24 @@ export class TS3Listener extends events.EventEmitter {
      * - a mapping of the TS-UID to the Discord-username is created
      */
     private async tagUp(g: discord.Guild, commander: Commander) {
-        let displayname = commander.getTS3DisplayName();
-        log("info", `Tagging up ${displayname} in ${g.name}.`);
+        log("info", `Tagging up ${(commander.getTS3DisplayName())} in ${g.name}.`);
         const registration = this.botgartClient.registrationRepository.getUserByAccountName(commander.getAccountName());
-        if (registration) {
+
+        let duser: discord.GuildMember | undefined;
+        if (registration !== undefined) {
             // the commander is member of the current discord -> give role
-            const crole = g.roles.cache.find(r => r.name === this.commanderRole);
-            const duser: discord.GuildMember | undefined = await g.members.fetch(registration.user); // cache.find(m => m.id === registration.user);
+            duser = await g.members.fetch(registration.user); // cache.find(m => m.id === registration.user);
             if (duser === undefined) {
                 log("warning", `Tried to find GuildMember for user with registration ID ${registration.user}, but could not find any. Maybe this is a caching problem?`)
             }
             commander.setDiscordMember(duser);
-            if (crole && commander.getDiscordMember()) {
-                await commander.getDiscordMember()?.roles.add(crole);
-            }
-            displayname = `${displayname} (${registration.registration_role})`;
+
+            await this.tagUpAssignRole(g, commander);
```

Comment:
this and `tagDownRemoveRole` are probably two of the very rare instances where asynchronicity comes in handy, as it really does not matter when the role is assigned/ removed. So removing `await` in these two cases is probably acceptable.

Reply:
I thougt about that, but decided that, because of the modification of the "commander" instance this should be waited for, as the commander object is part of the event, that will be emitted.